sql/src/test/java/io/crate/integrationtests/KillIntegrationTest.java

```diff
@@ -71,6 +71,7 @@ public void testKillInsertFromSubQuery() throws Exception {
                 ") with (number_of_replicas=1)");
         ensureYellow();
         assertGotCancelled("insert into new_employees (select * from employees)", null, true);
+        waitUntilShardOperationsFinished();
```

Comment:
maybe add a note that there could still be inflight upsert requests after the kill happened and that it is why it's waiting here for them to finish